src/System.Text.Primitives/System/Text/InvariantParser_ulong.cs

```diff
@@ -30,20 +30,23 @@ public static bool TryParse(Utf8String utf8Text, out ulong value, out int bytesC
                         return true;
                     }
                 }
-                try
+                else if (value > UInt64.MaxValue / 10) // overflow
                 {
-                    ulong candidate = checked(value * 10 + nextByte - '0');
```

Comment:
We might see slight improvements. I think ideally the JIT would recognize this common subexpression and optimize it out. But it probably can't coordinate that with the initial range check I mentioned above, so there's probably a bit of a benefit.